Assets/Scripts/Game/PlayerActivate.cs

```diff
@@ -91,20 +91,42 @@ public static bool IsBuildingOpen(DFLocation.BuildingTypes buildingType)
                     closeHours[(int)buildingType] > DaggerfallUnity.Instance.WorldTime.Now.Hour);
         }
 
-        // Allow mods to register custom model activation methods.
-        public delegate void ModelActivation(Transform transform);
-        private static Dictionary<string, ModelActivation> customModelActivations = new Dictionary<string, ModelActivation>();
+        // Allow mods to register custom flat / model activation methods.
+        public delegate void CustomActivation(Transform transform);
+        private static Dictionary<string, CustomActivation> customActivations = new Dictionary<string, CustomActivation>();
 
-        public static bool RegisterModelActivation(uint modelID, ModelActivation modelActivation)
+        public static void RegisterCustomActivation(uint modelID, CustomActivation customActivation)
         {
             string goModelName = GameObjectHelper.GetGoModelName(modelID);
-            DaggerfallUnity.LogMessage("RegisterModelActivation: " + goModelName, true);
-            if (!customModelActivations.ContainsKey(goModelName))
-            {
-                customModelActivations.Add(goModelName, modelActivation);
-                return true;
-            }
-            return false;
+            HandleRegisterCustomActivation(goModelName, customActivation);
+        }
+
+        public static void RegisterCustomActivation(int textureArchive, int textureRecord, CustomActivation customActivation)
+        {
+            string goFlatName = GameObjectHelper.GetGoFlatName(textureArchive, textureRecord);
+            HandleRegisterCustomActivation(goFlatName, customActivation);
+        }
+
+        private static void HandleRegisterCustomActivation(string goFlatModelName, CustomActivation customActivation)
+        {
+            DaggerfallUnity.LogMessage("HandleRegisterCustomActivation: " + goFlatModelName, true);
+            customActivations[goFlatModelName] = customActivation;
+        }
+
+        public static bool HasCustomActivation(uint modelID)
+        {
+            string goModelName = GameObjectHelper.GetGoModelName(modelID);
+            return CheckCustomActivation(goModelName);
+        }
+
+        public static bool HasCustomActivation(int textureArchive, int textureRecord)
+        {
+            string goFlatName = GameObjectHelper.GetGoFlatName(textureArchive, textureRecord);
+            return CheckCustomActivation(goFlatName);
+        }
+
+        public static bool CheckCustomActivation(string goFlatModelName) {
```

Comment:
I think is a bit confusing that this method is also public but has a different but similar name, especially because there is no xml summary. I personally would only use one name with three overloads.

Reply:
Hazelnut raised the same point in a conversation we had but I decided to put it out as is and collect more feedback. 

I have renamed `CheckCustomActivation` to `HasCustomActivation` because looking at it again, you guys have a point.